Rubberduck.RibbonDispatcher/Rubberduck.ExcelRibbon2013/CustomAppRibbonViewModel.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Office.Core;
+
+namespace Rubberduck.RibbonDispatcher.Excel2013 {
+    [ComVisible(true)][CLSCompliant(true)]
+    public class RubberduckRibbon : AbstractRibbon, IRibbonExtensibility {
+
+        public CustomAppRibbonViewModel() : base() {;}
+
+        public string GetCustomUI(string RibbonID) => GetResourceText("RubberDuck.RibbonSupport.CustomAppRibbon.xml");
+
+        [SuppressMessage("Microsoft.Design", "CA1061:DoNotHideBaseClassMethods", 
+            Justification = "This is simply how the Ribbon design works to share the Callback methods between multiple Office products & documents.")]
+        [CLSCompliant(false)]
+        public override void OnRibbonLoad(IRibbonUI ribbonUI) {
+            base.OnRibbonLoad(ribbonUI);
+
+            _standardButtonsViewModel = new StandardButtonsViewModel(RibbonFactory, (s,e) => _customButtonsViewModel.SetVisible(e.IsPressed));
+            _customButtonsViewModel   = new CustomButtonsViewModel(RibbonFactory);
+        }
+
+        StandardButtonsViewModel     _standardButtonsViewModel;
+        CustomButtonsViewModel       _customButtonsViewModel;
+
+        #region Helpers
+
+        private static string GetResourceText(string resourceName) {
```

Comment:
why is this method not relevant to other Ribbons?

Reply:
This is robo-code generated by Visual Studio as part of adding a ribbon template, that I have left as-is including the region.